src/components/inputName.components.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {useState} from "react";
+import "./component.styles.css";
+
+export function InputName(props: { placeholder: any; name: any, type: any, value: any, onChange: any}) {
+
+    const [inputName, setInputName] = useState("");
+
+    function handleChange   (event: { target: { value: any; name: any }; }){
+        setInputName(event.target.value);
+        props.onChange(event.target.name, event.target.value);
+    }
+   
+    return(
+        <div>
+            <input
+            className="input"
+            placeholder={props.placeholder}
+            name={props.name}
+            type="text"
+            value={props.value ? props.value : inputName}
+            onChange={(event) => handleChange(event)}
```

Comment:
Uma dica aqui: propriedades que recebem callbacks (esse lance de `() => {}`) podem ser escritas de forma mais rápida.

Eu não falei sobre isso antes porque queria um exemplo mais visual e é o que temos aqui.

Antes a gente fazia `onClick={handleClick}` que substituia o `onClick={() => handleClick()}`. A gente pode interpretar da seguinte forma: "quando o botão for pressionado, eu vou chamar uma função anônima que 'recebe um parâmetro vazio' (em outras palavras, não recebe nenhum parâmetro) e executa uma função `handleClick` que recebe nenhum parâmetro". Levando essa comparação ao "pé da letra", podemos dizer que "a propriedade `onChange` chama uma função anônima que recebe um parâmetro chamado `event` e executa uma outra função `handleChange` que recebe esse parâmetro `event`". Sendo assim, podemos fazer a mesma coisa que fazemos com a `handleClick`, fazendo apenas `onChange={handleChange}`.

Dizendo de outra forma, podemos cortar um caminho e passar direto uma função que tem como essência um callback (recebe um algo e executa algo em cima dele). Uma vez que essas propriedades esperam funções callbacks. É essa mesma [historia](https://github.com/LucasGobatto/Marcador-de-Pontos/pull/5#discussion_r517485591)

Reply:
Eu tava suspeitando que era a mesma coisa, mas como nos vídeos os caras sempre colocavam essas parada, achei melhor manter só pra garantir que ia funcionar.

Reply:
Mas quando eu altero isso no código, dá erro no componente no arquivo App.tsx